### PR TITLE
[Doctests] Make roberta-like meaningfull

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This makes RoBERTa-like doc examples meaningful

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/utils/doc.py [hunk 1]
```diff
@@ -172,28 +186,31 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
     >>> from transformers import {processor_class}, {model_class}
     >>> import torch
 
-    >>> torch.manual_seed(0)  # doctest: +IGNORE_RESULT
-
     >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
     >>> model = {model_class}.from_pretrained("{checkpoint}")
 
     >>> question, text = "Who was Jim Henson?", "Jim Henson was a nice puppet"
```
Comment (ydshieh) [inline src/transformers/utils/doc.py:192]: I have to admit that I don't know who Jim Henson is before google 😭. 
Personally, I would like to have examples that I know **very clearly** the model gives correct answer.
But no a big deal.

**Comment by patrickvonplaten** [comment]
> Thank you, @patrickvonplaten ! I do have one question: what should we do if other models give **not so good** results on these examples? We just try and use other input strings (which will give good outputs), and put the doc examples inside the model file?

Very good question! I think the nice tendency we have is:

- the more a model architecture is used, the higher the chance that there is a checkpoint that gives good results

So this gives us a good incentive to start with model architectures that are highly used: BERT, Eletrca, etc... once we come to model architectures that are less used we can think about adapted the template in `utils.doc.py` to allow the user to pass any input string through the decorator :-)

File: src/transformers/models/roberta/modeling_roberta.py [hunk 2]
```diff
@@ -1372,9 +1376,11 @@ def __init__(self, config):
     @add_start_docstrings_to_model_forward(ROBERTA_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
     @add_code_sample_docstrings(
         processor_class=_TOKENIZER_FOR_DOC,
-        checkpoint=_CHECKPOINT_FOR_DOC,
+        checkpoint="Jean-Baptiste/roberta-large-ner-english",
```
Comment (sgugger) [inline src/transformers/models/roberta/modeling_roberta.py:1379]: This looks like a non-organization checkpoint. Do we trust the user to leave the model as is in the future?

File: src/transformers/utils/doc.py [hunk 3]
```diff
@@ -204,41 +223,53 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
     >>> import torch
     >>> from transformers import {processor_class}, {model_class}
 
-    >>> torch.manual_seed(0)  # doctest: +IGNORE_RESULT
-
     >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
-    >>> model = {model_class}.from_pretrained("{checkpoint}", num_labels=2)
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
```
Comment (sgugger) [inline src/transformers/utils/doc.py:227]: This will break all classification examples apart from RoBERTa that just fixed, no?

It's also the most common mistake people do with sequence classification labels, so we should show a value here (whatever value it is).

File: src/transformers/utils/doc.py [hunk 4]
```diff
@@ -204,41 +223,53 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
 
     ```python
     >>> import torch
     >>> from transformers import {processor_class}, {model_class}
 
-    >>> torch.manual_seed(0)  # doctest: +IGNORE_RESULT
-
     >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
-    >>> model = {model_class}.from_pretrained("{checkpoint}", problem_type="multi_label_classification", num_labels=2)
+    >>> model = {model_class}.from_pretrained("{checkpoint}", problem_type="multi_label_classification")
```
Comment (sgugger) [inline src/transformers/utils/doc.py:253]: Same here.

**Review by sgugger — APPROVED** [review]
Thanks for working on those. We shouldn't remove the `num_labels` argument entirely in the sequence classification example IMO, as it's very important to show users how to set it. We can add a comment to let the user know they can add it if we really have to remove it from the code.

File: src/transformers/models/roberta/modeling_roberta.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/transformers/models/roberta/modeling_roberta.py:1379]: Our tests will tell I think! Think it's still better then having a checkpoint outputting random outputs

File: src/transformers/models/roberta/modeling_roberta.py [hunk 5]
```diff
@@ -1067,6 +1067,8 @@ def set_output_embeddings(self, new_embeddings):
         output_type=MaskedLMOutput,
         config_class=_CONFIG_FOR_DOC,
         mask="<mask>",
+        expected_output="' Paris'",
```
Comment (patrickvonplaten) [inline src/transformers/models/roberta/modeling_roberta.py:1070]: We should only have to add these values to the modeling file to make the doc test pass for other RoBERTa like tests :-)

File: utils/documentation_tests.txt [hunk 6]
```diff
@@ -1,21 +1,35 @@
...
+src/transformers/models/mbart/modeling_mbart.py
 src/transformers/models/mbart/modeling_mbart.py
 src/transformers/models/pegasus/modeling_pegasus.py
+src/transformers/models/pegasus/modeling_pegasus.py
+src/transformers/models/plbart/modeling_plbart.py
 src/transformers/models/plbart/modeling_plbart.py
 src/transformers/models/poolformer/modeling_poolformer.py
 src/transformers/models/resnet/modeling_resnet.py
+src/transformers/models/resnet/modeling_resnet.py
+src/transformers/models/roberta/modeling_roberta.py
```
Comment (patrickvonplaten) [inline utils/documentation_tests.txt:32]: and let's not forgot to add the code file here
